includes/class-micropub-endpoint.php

```diff
@@ -140,14 +166,39 @@ protected static function load_input( $request ) {
 	 * Check scope
 	 *
 	 * @param string $scope
+	 * @param id $user_id. If supplied will check user permissions
```

Comment:
it took me a while to understand the logic here, ie:
* if we can map the token to a user, check the scope against the user's permissions
* otherwise, assume we can do anything, and just check the token's scopes themselves

mind adding that to the comment here? or in your own words if i got it wrong?